### DISCUSSION TITLE
Getting mesh topology from the pipeline

### DISCUSSION THREAD

**Discussion by panthavma** [bnpr/Malt#291] [Q&A] [answered]
Hello! I want to access the mesh's data from a custom pipeline. I need to have both the vertices' position and normal, but also the adjacency information if possible.

From what I gather compared to lights, we load the mesh directly from Blender through BlenderMalt and CBlenderMalt, and the bridge will load it up for us in VBOs with MeshCustomLoad. In my case however, I need access to it on the CPU side.

Is there a way to get those values cleanly, without reading back from the GPU or recompiling Malt? And if I have to recompile Malt, do you have tips on how to do it in a nice way?

**Comment by pragma37** [comment, marked as answer]
It's not doable at the moment, but I think this should be supported. I'll look into it.

In any case, you wouldn't have to recompile Malt to do this since only requires changes on the Python side.

Adjacency information is not extracted at the moment, but I'm currently prototyping geometry based lines, so it may be included in the near future.

Do you have any related requests?

**Reply by panthavma** [reply]
Actually that's exactly what I'm doing. I just started a PhD in expressive rendering and my first subject is geometry based lines like freestyle!
If so, it might be interesting to have an extended discussion, as we seem to be working towards the same goal. Would you be available for that ?

**Reply by pragma37** [reply]
Sure!

**Reply by panthavma** [reply]
I sent you a message on discord, let's organize that!

**Reply by panthavma** [reply]
For future reference: https://github.com/bnpr/Malt/commit/3b05b9025b1f5e8993013bc5919837c53d4ef554
